### PR TITLE
adding secretsmanager support to templates

### PR DESCRIPTION
**Pull request by jhodnett2**
Adding a basic secret to the template.  Also providing a templated bash script that will load hidden.env into that secret.

Minor instructions are still needed in the blog post to use this.

### CODE REVIEW

File: env/dev/load-secrets.tpl [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+}
+
+AWS_PROFILE=${aws_profile}
+AWS_DEFAULT_REGION=${region}
+NAME=${secret}
+export AWS_PROFILE  AWS_DEFAULT_REGION
+
+if [ -e "./hidden.env" ]; then
+
+   FIND=$(aws secretsmanager describe-secret --secret-id $NAME)
```
Comment (abarfoot) [inline env/dev/load-secrets.tpl]: Consider setting $AWS_DEFAULT_OUTPUT to json or text (whichever you expect)

File: env/dev/load-secrets.tpl [hunk 2]
```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
```
Comment (abarfoot) [inline env/dev/load-secrets.tpl]: Maybe rename the script?   I initially thought that a script that "loads secrets" would *read* secrets from SecretsManager.  It's not clear that it's a destructive/overwriting update to SecretsManager.

**Comment by abarfoot** [comment]
Add hidden.env to .gitignore ?

**Comment by abarfoot** [comment]
[Sorry for closing/re-opening; I clicked the wrong button.]

**Comment by jhodnett2** [comment]
As we talked about, there should never be a hidden.env in the context of this repo.  It will be copied to users.  But in file/new scenarios it will have to be addressed (either via John's script, or good blog post).

**Pull request closed without merging**